File: src/lib/categories.ts
import type { CategoryId } from "./types";

export interface CategoryMeta {
  id: CategoryId;
  label: string;
  emoji: string;
  /** tailwind text + bg tint for chips */
  tint: string;
  /** hex used for charts */
  color: string;
}

export const CATEGORIES: CategoryMeta[] = [
  { id: "mobile", label: "Mobile", emoji: "📱", tint: "bg-sky-50 text-sky-700", color: "#0ea5e9" },
  { id: "broadband", label: "Broadband", emoji: "🌐", tint: "bg-indigo-50 text-indigo-700", color: "#6366f1" },
  { id: "ott", label: "OTT & Streaming", emoji: "🎬", tint: "bg-rose-50 text-rose-700", color: "#f43f5e" },
  { id: "insurance", label: "Insurance", emoji: "🛡️", tint: "bg-emerald-50 text-emerald-700", color: "#10b981" },
  { id: "credit-card", label: "Credit Card", emoji: "💳", tint: "bg-violet-50 text-violet-700", color: "#8b5cf6" },
  { id: "loan-emi", label: "Loan EMI", emoji: "🏦", tint: "bg-amber-50 text-amber-700", color: "#f59e0b" },
  { id: "electricity", label: "Electricity", emoji: "⚡", tint: "bg-yellow-50 text-yellow-700", color: "#eab308" },
  { id: "gas", label: "Gas", emoji: "🔥", tint: "bg-orange-50 text-orange-700", color: "#f97316" },
  { id: "school-fees", label: "School Fees", emoji: "🎓", tint: "bg-teal-50 text-teal-700", color: "#14b8a6" },
  { id: "gym", label: "Gym & Fitness", emoji: "🏋️", tint: "bg-lime-50 text-lime-700", color: "#84cc16" },
  { id: "software", label: "Software", emoji: "🧩", tint: "bg-cyan-50 text-cyan-700", color: "#06b6d4" },
  { id: "membership", label: "Memberships", emoji: "⭐", tint: "bg-pink-50 text-pink-700", color: "#ec4899" },
];

const byId = new Map(CATEGORIES.map((c) => [c.id, c]));

export function category(id: CategoryId): CategoryMeta {
  return byId.get(id) ?? CATEGORIES[CATEGORIES.length - 1];
}

File: flutter/okna/lib/data/categories.dart
import 'package:flutter/material.dart';
import '../models/subscription.dart';

class CategoryMeta {
  final CategoryId id;
  final String label;
  final String emoji;
  final Color color;
  const CategoryMeta(this.id, this.label, this.emoji, this.color);

  Color get tintBg => color.withValues(alpha: 0.12);
  Color get tintFg => Color.alphaBlend(color.withValues(alpha: 0.85), Colors.black);
}

const List<CategoryMeta> kCategories = [
  CategoryMeta(CategoryId.mobile, 'Mobile', '📱', Color(0xFF0EA5E9)),
  CategoryMeta(CategoryId.broadband, 'Broadband', '🌐', Color(0xFF6366F1)),
  CategoryMeta(CategoryId.ott, 'OTT & Streaming', '🎬', Color(0xFFF43F5E)),
  CategoryMeta(CategoryId.insurance, 'Insurance', '🛡️', Color(0xFF10B981)),
  CategoryMeta(CategoryId.creditCard, 'Credit Card', '💳', Color(0xFF8B5CF6)),
  CategoryMeta(CategoryId.loanEmi, 'Loan EMI', '🏦', Color(0xFFF59E0B)),
  CategoryMeta(CategoryId.electricity, 'Electricity', '⚡', Color(0xFFEAB308)),
  CategoryMeta(CategoryId.gas, 'Gas', '🔥', Color(0xFFF97316)),
  CategoryMeta(CategoryId.schoolFees, 'School Fees', '🎓', Color(0xFF14B8A6)),
  CategoryMeta(CategoryId.gym, 'Gym & Fitness', '🏋️', Color(0xFF84CC16)),
  CategoryMeta(CategoryId.software, 'Software', '🧩', Color(0xFF06B6D4)),
  CategoryMeta(CategoryId.membership, 'Memberships', '⭐', Color(0xFFEC4899)),
];

final Map<CategoryId, CategoryMeta> _byId = {for (final c in kCategories) c.id: c};

CategoryMeta categoryMeta(CategoryId id) => _byId[id] ?? kCategories.last;
